src/distributions/rayon.rs

```diff
@@ -0,0 +1,212 @@
+    where D: Distribution<T> + Send + Sync,
+          R: Rng + SeedableRng + Send,
+          T: Send,
+{
+    fn drive<C>(self, consumer: C) -> C::Result
+        where C: Consumer<Self::Item>
+    {
+        bridge(self, consumer)
+    }
+
+    fn len(&self) -> usize {
+        self.amount
+    }
+
+    fn with_producer<CB>(self, callback: CB) -> CB::Output
+        where CB: ProducerCallback<Self::Item>
+    {
+        callback.callback(
+            DistProducer {
+                distr: self.distr.clone(),
+                amount: self.amount,
+                rng: self.rng,
+                phantom: ::core::marker::PhantomData,
+            }
+        )
+    }
+}
+
+/// FIXME
+#[cfg(feature = "rayon")]
+#[derive(Debug)]
+pub struct DistProducer<'a, D: 'a, R, T> {
+    distr: &'a D,
+    rng: R,
+    amount: usize,
+    phantom: ::core::marker::PhantomData<T>,
+}
+
+/// This method is intented to be used by fast and relatively simple PRNGs used
+/// for simulations etc. While it will also work with cryptographic RNGs, that
+/// is not optimal.
+///
+/// Every time `rayon` splits the work in two to create parallel tasks, one new
+/// PRNG is created. The original PRNG is used to seed the new one using
+/// `SeedableRng::from_rng`. **Important**: Not all RNG algorithms support this!
+/// Notably the low-quality plain Xorshift, the current default for `SmallRng`,
+/// will simply clone itself using this method instead of seeding the split off
+/// RNG well. Consider using something like PCG or Xoroshiro128+.
+///
+/// It is hard to predict what will happen to the statistical quality of PRNGs
+/// when they are split off many times, and only very short runs are used. We
+/// limit the minimum number of items that should be used of the PRNG to at
+/// least 100 to hopefully keep similar statistical properties as one PRNG used
+/// continuously.
+#[cfg(feature = "rayon")]
+impl<'a, D, R, T> Producer for DistProducer<'a, D, R, T>
+    where D: Distribution<T> + Send + Sync,
+          R: Rng + SeedableRng + Send,
+          T: Send,
+{
+    type Item = T;
+    type IntoIter = BoundedDistIter<'a, D, R, T>;
+    fn into_iter(self) -> Self::IntoIter {
+        BoundedDistIter {
+            distr: self.distr,
+            amount: self.amount,
+            rng: self.rng,
+            phantom: ::core::marker::PhantomData,
+        }
+    }
+
+    fn split_at(mut self, index: usize) -> (Self, Self) {
+        assert!(index <= self.amount);
+        // Create a new PRNG of the same type, by seeding it with this PRNG.
+        // `from_rng` should never fail.
+        let new = DistProducer {
+            distr: self.distr,
+            amount: self.amount - index,
+            rng: R::from_rng(&mut self.rng).unwrap(),
+            phantom: ::core::marker::PhantomData,
+        };
+        self.amount = index;
+        (self, new)
+    }
+
+    fn min_len(&self) -> usize {
+        100
+    }
+}
+
+/// FIXME
```

Comment:
Fix what?

Reply:
It does not compile without doc comments. Maybe a nice place to explain why we sample an `amount`. I believe we can't just keep generating values until the user decides it has enough; Rayon seems to (reasonably) not support much functionality when the number of values is unknown.